packages/neo-one-smart-contract-lib/src/ownership/Ownable.ts

```diff
@@ -0,0 +1,52 @@
+import { Address, constant, createEventNotifier, SmartContract } from '@neo-one/smart-contract';
+
+/**
+ * @title Ownable
+ * @dev A ownable contract has an owner address
+ */
+
+/* tslint:disable-next-line: variable-name */
+const transferred_ownership = createEventNotifier<Address | undefined, Address>('transferred ownership', 'from', 'to');
+
+export function Ownable<TBase extends Constructor<SmartContract>>(base: TBase) {
+  class OwnableClass extends base {
+    protected mutableOwner: Address | undefined;
+    public constructor(...args: any[]) {
+      super();
+      this.mutableOwner = args[0];
+    }
+
+    @constant
+    public get owner(): Address | undefined {
```

Comment:
This seems a bit awkward making it possibly `undefined`. Does it make sense to enforce there is always an owner for something that is `Ownable`?

Reply:
@dicarlo2  The template for this framework has a `renounce()` that sets `_owner` to `Address(0)`.
Is it possible for address zero to make requests to the blockchain?   Otherwise, what `  Address Literal String ` would be best to use for this `renounced` state?

Reply:
Hmm, I think in that case if we want to support renouncing then we should leave it as is. We could set it to an address with all 0s, but I think setting it to undefined makes more sense idiomatically for typescript.

Instead, let's add a method that's like `ownerOrThrow` which returns an `Address` to cover the likely common pattern of a user expecting the contract to always have an owner.

Reply:
👍